src/fava_portfolio_returns/core/portfolio.py

```diff
@@ -79,6 +81,7 @@ def __init__(
             entries, dcontext, self.beangrow_cfg, entries[-1].date, False, beangrow_debug_dir
         )
         self.investment_groups = group_investments(self.beangrow_cfg, self.account_data_map)
+        self.all_commodities = get_ledger_commodities(entries)
```

Comment:
Instead of adding another grouping (accounts, groups, beangrow currencies, ledger currencies), I'm inclined to replace the current code which extracts currencies:
https://github.com/andreasgerstmayr/fava-portfolio-returns/blob/c46c1945d0b6f5a1f075c650e882ac74a826e8f0/src/fava_portfolio_returns/core/portfolio.py#L249-L260
with your `get_ledger_commodities` function.

What do you think? Is there any requirement to have two different sets of currencies, one from the beangrow config and one from the ledger?

As far as I see it, in all cases the ledger currencies are a superset of the investment currencies, so I'd just replace the list of beangrow currencies with the list of ledger currencies.

Reply:
> Although, looking at your specific suggestion, the problem with the complete replacement is that currencies will also get displayed in the portfolio filter – which is a slight regression in experience. That is, currencies that don't participate in portfolio costs will always result in filtering down to an empty portfolio.

Good point!

> Maybe the best solution is to leave both, just use only all_commodities in the "Compare with" dropdown, but without a different grouping?

I would suggest having a single list of currencies internally (with a property `isInvestment`), and filtering them in the `<InvestmentsSelection>` component (using the `type` prop, by adding a new type `InvestmentCurrency` to the enum).